src/nifs/protogalaxy/mod.rs

```diff
@@ -37,21 +37,58 @@ pub struct ProtoGalaxy<C: CurveAffine, const L: usize> {
     _marker: PhantomData<C>,
 }
 
-impl<C: CurveAffine, const L: usize> ProtoGalaxy<C, L> {
+pub(crate) struct Challenges<F: PrimeField> {
+    pub delta: F,
+    pub alpha: F,
+    pub gamma: F,
+}
+
+impl<F: PrimeField> Challenges<F> {
     #[instrument(skip_all)]
-    pub(crate) fn generate_challenge<'i, RO: ROTrait<C::Base>>(
-        pp_digest: &C,
+    pub(crate) fn generate_one<'i, RO: ROTrait<C::Base>, C: CurveAffine<Base = F>>(
```

Comment:
what does `generate_one` mean?

Reply:
We have scenarios where we use everything except `poly_{F,K}` for challenge. For example for the first `beta` within initial accumulator or `delta` within `nifs::protogalaxy::prove`. It's not the best name, but I haven't found another to indicate that `protogalaxy::Challenges::generate` is to generate all the `challenge` needed for protogalaxy, and `generate_one` is to generate only one of them based on all 'prefix' data.